### PR TITLE
Combobox: add loose-match option for matching query

### PR DESCRIPTION
**Pull request by NazimHAli**
**This change:** (check at least one)

- [x] Adds a new feature
- [ ] Fixes a bug
- [ ] Improves maintainability
- [ ] Improves documentation
- [ ] Is a release activity

**Is this a breaking change?** (check one)

- [ ] Yes
- [x] No

**Is the:** (complete all)

- [x] Title of this pull request clear, concise, and indicative of the issue number it addresses, if any?
- [x] Test suite(s) passing?
- [x] Code coverage maximal?
- [x] Changeset added?
- [x] Component status page up to date?

**What does this change address?**
Provides the option to loosely match user input values to the text options. By default there's an exact match and this excludes cases when a query or text has accents or different casing.

**How does this change work?**
Added method to normalize strings using https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/normalize#parameters

**Additional context**
Users couldn't find their institutions when searching on https://www.jstor.org/site/collection-list/ because the official spelling contains an accent.

### CODE REVIEW

File: packages/pharos/src/components/combobox/pharos-combobox.ts [hunk 1]
```diff
@@ -435,6 +448,13 @@ export class PharosCombobox extends ScopedRegistryMixin(FormMixin(FormElement))
     this.open = !this.open;
   }
 
+  private _normalizeString(textString: string) {
+    return textString
+      .normalize('NFKD')
+      .replace(/[\u0300-\u036f]/g, '')
+      .toLowerCase();
+  }
```
Comment (NazimHAli) [inline packages/pharos/src/components/combobox/pharos-combobox.ts:456]: There's also the method `localeCompare` that filters accents - testing it now and might removed the need for this method.

File: packages/pharos/src/components/combobox/pharos-combobox.ts [hunk 2]
```diff
@@ -435,6 +448,13 @@ export class PharosCombobox extends ScopedRegistryMixin(FormMixin(FormElement))
     this.open = !this.open;
   }
 
+  private _normalizeString(textString: string) {
+    return textString
+      .normalize('NFKD')
+      .replace(/[\u0300-\u036f]/g, '')
```
Comment (daneah) [inline packages/pharos/src/components/combobox/pharos-combobox.ts:454]: Might you leave a comment here explaining this? I don't know offhand what it does, myself! Is that a range of unprintable unicode characters?

**Review by daneah — APPROVED** [review]
This looks really nice, thank you for adding it! I have one suggestion for future maintainability

File: packages/pharos/src/components/combobox/pharos-combobox.ts (hunk 1 above)
Comment (NazimHAli) [reply packages/pharos/src/components/combobox/pharos-combobox.ts:456]: `localeCompare` is for a different use case, not partial matching.

File: packages/pharos/src/components/combobox/pharos-combobox.ts (hunk 2 above)
Comment (NazimHAli) [reply packages/pharos/src/components/combobox/pharos-combobox.ts:454]: It's a range of all possible diacritics - after text is normalized, than we can remove the characters and compare the text.

File: packages/pharos/src/components/combobox/pharos-combobox.ts (hunk 1 above)
Comment (NazimHAli) [reply packages/pharos/src/components/combobox/pharos-combobox.ts:456]: Testing variations of this method, there are `search, sensitivity` options https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator/Collator#options

Comment (NazimHAli) [reply packages/pharos/src/components/combobox/pharos-combobox.ts:465]: The search option won't do partial matching either.

```js
"nèat".localeCompare("neat now", { usage: "search", ignorePunctuation: true, sensitivity: "base" })
```

File: packages/pharos/src/components/combobox/pharos-combobox.ts (hunk 2 above)
Comment (daneah) [reply packages/pharos/src/components/combobox/pharos-combobox.ts:463]: Please leave this as a code comment, if you don't mind!

**Review by brentswisher — APPROVED** [review]
Nice work! I had struggled a little with this in the past (In Oracle, not JS) so I was curious and this looks really good. I tried this with some cyrillic characters that caused issues then and it handled them well. 

I did try some things which are a step further than removing the accent mark, like seeing if the Germanic ß matches "ss" and that didn't appear to work - but I don't think that is necessary here. 

Just mentioning this because I find it interesting, I don't think it is worth adding a package dependency to fix.  This article goes into detail about it, but essentially `normalize` takes off the extra unicode diacritic and leaves the base letter, and there are some weird edge cases like `ß` and `ł` where that doesn't work: https://javascript.plainenglish.io/not-so-obvious-removal-of-diacritics-in-javascript-explained-and-done-right-52f4aeb3c85

**Comment by NazimHAli** [comment]
> Nice work! I had struggled a little with this in the past (In Oracle, not JS) so I was curious and this looks really good. I tried this with some cyrillic characters that caused issues then and it handled them well.
> 
> I did try some things which are a step further than removing the accent mark, like seeing if the Germanic ß matches "ss" and that didn't appear to work - but I don't think that is necessary here.
> 
> Just mentioning this because I find it interesting, I don't think it is worth adding a package dependency to fix. This article goes into detail about it, but essentially `normalize` takes off the extra unicode diacritic and leaves the base letter, and there are some weird edge cases like `ß` and `ł` where that doesn't work: https://javascript.plainenglish.io/not-so-obvious-removal-of-diacritics-in-javascript-explained-and-done-right-52f4aeb3c85

Thank you for sharing! As we increase the amount of content, I think we'll get to a point where we will need to address the cases you mentioned. We already have some collections in Arabic, Farsi, Hindi/Urdu/Benghali, (I don't know offhand if we have content in Germanic, but assume we do because of European collections) that need to be searchable.

**Pull request merged**